src/rdf4cpp/rdf/parser/RDFFileParser.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef RDF4CPP_RDFFILEPARSER_HPP
+#define RDF4CPP_RDFFILEPARSER_HPP
+
+#include <fstream>
+#include "IStreamQuadIterator.hpp"
+
+namespace rdf4cpp::rdf::parser {
+    /**
+     * Similar to rdf4cpp::rdf::parser::IStreamQuadIterator
+     * Parses the file by the given path and tries to extract Quads given in TURTLE format.
+     *
+     * @note the iterator _starts on_ the first Quad
+     * @note An exhausted iterator becomes the end-of-stream iterator.
+     * @warning The value pointed to by an end-of-stream iterator is undefined
+     *
+     * @example
+     * @code
+     * for (const auto& e : rdf4cpp::rdf::parser::RDFFileParser{"somewhere.ttl"}) {
+     *  if (e.has_value())
+     *     std::cout << e.value();
+     *  else
+     *     std::cout << e.error();
+     * }
+     * @endcode
+     */
+    class RDFFileParser {
+        std::string filePath_;
+        ParsingFlags flags_;
+        storage::node::NodeStorage node_storage_;
+
+    public:
+        explicit RDFFileParser(const std::string& filePath, ParsingFlags flags = ParsingFlags::none(),
```

Comment:
please use snake_case for instance names (no matter if variable, parameter or static value.